designcompose/src/main/java/com/android/designcompose/Layout.kt

```diff
@@ -541,81 +568,116 @@ internal inline fun DesignFrameLayout(
-                val layoutData = placeable.designLayoutData
-                if (layoutData == null) {
-                    // A null layout should only happen if the child is a node that uses one of
-                    // Compose's built in layouts such as Row, Column, or LazyGrid.
-                    placeable.place(0, 0)
-                    if (index != 0) Log.d(TAG, "Place error no layoutData: $name index $index")
+    var myLayout = LayoutManager.getLayoutWithDensity(layoutId)
+    if (myLayout == null)
+        Log.d(TAG, "designMeasurePolicy error: null layout $name layoutId $layoutId")
+    // Get width and height from constraints if they are fixed. Otherwise get them from layout.
+    val myWidth =
+        if (constraints.hasFixedWidth && constraints.maxWidth != 0) constraints.maxWidth
+        else myLayout?.width() ?: 0
+    val myHeight =
+        if (constraints.hasFixedHeight && constraints.maxHeight != 0) constraints.maxHeight
+        else myLayout?.height() ?: 0
+
+    // If constraints have forced a size that does not match our layout size, set this size into
+    // the layout manager so that it can set this size in the Rust layout manager and then
+    // recalculate any layouts that many have changed. However, only do this if the layout state
+    // has not changed after computing child layouts. This is necessary in the case where
+    // calling setNodeSize on a node causes an ancestor to resize.
+    val layoutState2 = LayoutManager.getLayoutState(layoutId)
+    if (
+        layoutState == layoutState2 &&
+            myLayout != null &&
+            (myWidth != myLayout?.width() || myHeight != myLayout?.height())
+    )
+        LayoutManager.setNodeSize(layoutId, rootLayoutId, myWidth, myHeight)
+    layout(myWidth, myHeight) {
+        // Place children in the parent layout
+        placeables.forEachIndexed { index, placeable ->
+            val layoutData = placeable.designLayoutData
+            if (layoutData == null) {
+                // A null layout should only happen if the child is a node that uses one of
+                // Compose's built in layouts such as Row, Column, or LazyGrid.
+                placeable.place(0, 0)
+                if (index != 0) Log.d(TAG, "Place error no layoutData: $name index $index")
+            } else {
+                val childLayout = LayoutManager.getLayoutWithDensity(layoutData.layoutId)
+                if (childLayout == null) {
+                    Log.d(TAG, "Place error null layout: $name child $index layoutId $layoutId")
                 } else {
-                    val childLayout = LayoutManager.getLayoutWithDensity(layoutData.layoutId)
-                    if (childLayout == null) {
-                        Log.d(TAG, "Place error null layout: $name child $index layoutId $layoutId")
-                    } else {
-                        placeable.place(x = childLayout.left(), y = childLayout.top())
+                    var hScrollOffset = 0
```

Comment:
I was surprised to see that Compose seems to [implement the scroll offset using layout, too](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/Scroll.kt).